packages/mobile/src/backup/BackupQuiz.tsx

```diff
@@ -82,21 +84,32 @@ export class BackupQuiz extends React.Component<Props, State> {
     this.setBackSpace()
   }
 
-  componentDidMount() {
-    const mnemonic = this.getMnemonicFromNavProps()
+  componentDidMount = async () => {
+    const mnemonic = await this.retrieveMnemonic()
     const shuffledMnemonic = getShuffledWordSet(mnemonic)
     this.setState({
       mnemonicWords: shuffledMnemonic,
       mnemonicLength: shuffledMnemonic.length,
     })
   }
 
-  getMnemonicFromNavProps() {
-    const mnemonic = this.props.route.params.mnemonic
-    if (!mnemonic) {
-      throw new Error('Mnemonic missing from nav props')
+  retrieveMnemonic = async () => {
+    try {
+      const mnemonic = await getStoredMnemonic()
+      if (!mnemonic) {
+        throw new Error('Mnemonic not found in key store')
+      }
+      return mnemonic
+    } catch {
+      const message = 'Failed to retrieve mnemonic'
+      Logger.error('BackupQuiz/retrieveMnemonic', message)
+      this.props.showError(ErrorMessages.FAILED_FETCH_MNEMONIC)
+      CeloAnalytics.track(CustomEventNames.backup_error, {
+        title: message,
+        component: 'BackupQuiz',
+      })
+      return ''
```

Comment:
I do not understand why we return `''` in case of an error?

Reply:
basically that was just to get typescript to not complain about type being undefined. 

would love to talk about handling the error case cause at the moment feels like there are a lot of `if`s and `trys` and its getting out of hand